dspace-spring-rest/src/test/java/org/dspace/app/rest/MetadataSchemaRestRepositoryIT.java

```diff
@@ -7,50 +7,218 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.dspace.app.rest.builder.MetadataSchemaBuilder;
+import org.dspace.app.rest.converter.MetadataSchemaConverter;
 import org.dspace.app.rest.matcher.MetadataschemaMatcher;
+import org.dspace.app.rest.model.MetadataSchemaRest;
 import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
 import org.dspace.content.MetadataSchema;
 import org.hamcrest.Matchers;
 import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class MetadataSchemaRestRepositoryIT extends AbstractControllerIntegrationTest {
 
+    private static final String TEST_NAME = "testSchemaName";
+    private static final String TEST_NAMESPACE = "testSchemaNameSpace";
+
+    private static final String TEST_NAME_UPDATED = "testSchemaNameUpdated";
+    private static final String TEST_NAMESPACE_UPDATED = "testSchemaNameSpaceUpdated";
+
+    @Autowired
+    MetadataSchemaConverter metadataSchemaConverter;
+
     @Test
     public void findAll() throws Exception {
 
         context.turnOffAuthorisationSystem();
         MetadataSchema metadataSchema = MetadataSchemaBuilder.createMetadataSchema(context, "ATest", "ANamespace")
-                                                             .build();
+            .build();
 
         getClient().perform(get("/api/core/metadataschemas"))
-                   .andExpect(status().isOk())
-                   .andExpect(content().contentType(contentType))
-                   .andExpect(jsonPath("$._embedded.metadataschemas", Matchers.hasItem(
-                       MetadataschemaMatcher.matchEntry()
-                   )))
-                   .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/core/metadataschemas")))
-                   .andExpect(jsonPath("$.page.size", is(20)));
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$._embedded.metadataschemas", Matchers.hasItem(
+                        matchEntry()
+                )))
+                .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/core/metadataschemas")))
+                .andExpect(jsonPath("$.page.size", is(20)));
     }
 
-
     @Test
     public void findOne() throws Exception {
 
         context.turnOffAuthorisationSystem();
+        MetadataSchema metadataSchema = MetadataSchemaBuilder.createMetadataSchema(context, "ATest", "ANamespace")
+                .build();
+
```

Comment:
We should be turning back *on* the AuthorizationSystem here, shouldn't we?  We only need it off temporarily in order to create test data.